src/NHibernate/Impl/MultiQueryImpl.cs

```diff
@@ -526,7 +526,7 @@ protected List<object> DoList()
 
 			try
 			{
-				using (var reader = resultSetsCommand.GetReader(commandTimeout != RowSelection.NoValue ? commandTimeout : (int?)null))
+				using (var reader = resultSetsCommand.GetReader(_timeout))
```

Comment:
Breaking change here: users actively setting the timeout with `NoValue` (or `-1`) will have they code switched from a "use default timeout" behavior to "throw an `ArgumentException`". And so, for consistency, it would be better to put the `NoValue` logic into `MultiCriteriaImpl` too.